### PR TITLE
CRUD Basic

### PR DESCRIPTION
**Pull request by tuanminh9704**

### CODE REVIEW

File: controllers/game.controller.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+}
+
+//[POST] /games
+module.exports.createGame = (req, res) => {
+    try {
+        const newGame = req.body;
+
+        if(newGame.title == "" || newGame.genre == ""){
```
Comment (namvu97) [inline controllers/game.controller.js]: Validate a string using trim(). For example, if a user sends the string ' ' (with only spaces), trim() should remove the spaces.

File: controllers/game.controller.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+//[PATCH] /games/:id
+module.exports.updateGame = (req, res) => {
+    try {
+        const gameId = parseInt(req.params.id);
+        const {title, genre, price} = req.body;
+
+        const game = games.find(item => item.gameId === gameId);
+        if(title){
```
Comment (namvu97) [inline controllers/game.controller.js]: Similar with the above that need validate a string using trim().

File: controllers/game.controller.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            game.title = title;
+        }
+        if(genre){
```
Comment (namvu97) [inline controllers/game.controller.js]: Similar with the above that need validate a string using trim().

File: controllers/game.controller.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            game.genre = genre;
+        }
+        if(price){
```
Comment (namvu97) [inline controllers/game.controller.js]: Add validation to ensure that the price is a valid number (including a decimal point, if needed).

File: controllers/game.controller.js [hunk 5]
```diff
@@ -90,14 +91,23 @@ module.exports.updateGame = (req, res) => {
...
+        if(title.trim() === "" || genre.trim() == "" || isNaN(price) || price < 0){
+            res.json({
+                code: 500,
+                message: "title, genre or price is invalid"
+            })
         }
-        if(price){
-            game.price = price;
+        else{
+            if(title){
```
Comment (kichitto-dev1) [inline controllers/game.controller.js]: There's no need to check if title, genre, and price have values again; this was already verified above

**Pull request merged**